### PR TITLE
URI encoding of redirects

### PR DESCRIPTION
**Pull request by veloman-yunkan**
This is the first on of expected multiple fixes addressing kiwix/libkiwix#775.

This particular fix should at least eliminate @DarkmatterUAE's pain described at kiwix/kiwix-tools#589 :smiley:. I tested it with the smaller and newer [wikipedia_en_all_mini_2023-01.zim](http://download.kiwix.org/zim/wikipedia/wikipedia_en_all_mini_2023-01.zim) instead of [wikipedia_en_all_nopic_2022-01.zim](http://download.kiwix.org/zim/wikipedia/wikipedia_en_all_nopic_2022-01.zim) used in the original ticket, but they shouldn't be different with the respect to the data that triggered the bug.

Fixes kiwix/kiwix-tools#589

### CODE REVIEW

**Comment by kelson42** [comment]
@mgautierfr @veloman-yunkan Probably not related to this PR, but worried that the macOS CI fails.

File: test/data/corner_cases/wtf?.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
```
Comment (mgautierfr) [inline test/data/corner_cases/wtf?.html:1]: (Comment related to the commit, not the line)

Adding the 3 files to `corner_cases.zim` make the zim file increasing by 32.3 KB (1300%), where remove the fulltextindex was decreasing the file by 71KB. This seems too much. Can you verify that you have commit the right zim file ?

**Review by mgautierfr — CHANGES_REQUESTED** [review]

File: test/data/corner_cases/wtf?.html (hunk 1 above)
Comment (veloman-yunkan) [reply test/data/corner_cases/wtf?.html:1]: Everything is correct. So far `corner_cases.zim` contained only empty items which provided an opportunity to omitting the title index. Appearance of the first entry with a title resulted in the title index being generated.

**Comment by veloman-yunkan** [comment]
No changes, only rebased after merging #857.

**Comment by kelson42** [comment]
@mgautierfr Good to merge then?

File: test/data/corner_cases/wtf?.html (hunk 1 above)
Comment (mgautierfr) [reply test/data/corner_cases/wtf?.html:1]: Arf. This is a pitty. I wonder if we should not have a option to not create xapian title index at all.
(Or maybe use a library compiled without xapian)

Anyway, not for this PR

**Review by mgautierfr — APPROVED** [review]

**Pull request merged**